app/components/comment-item.js

```diff
@@ -1,5 +1,49 @@
 import Ember from 'ember';
 
 export default Ember.Component.extend({
-  classNames: ['comment-item']
+  classNames: ['comment-item'],
+  classNameBindings: ['isEditing:editing'],
+
+  session: Ember.inject.service(),
+
+  currentUserId: Ember.computed.alias('session.session.authenticated.user_id'),
+  commentAuthorId: Ember.computed.alias('comment.user.id'),
+  currentUserIsCommentAuthor: Ember.computed('currentUserId', 'commentAuthorId', function() {
+    let userId = parseInt(this.get('currentUserId'), 10);
+    let authorId = parseInt(this.get('commentAuthorId'), 10);
+    return userId === authorId;
+  }),
+
+  canEdit: Ember.computed.and('session.isAuthenticated', 'currentUserIsCommentAuthor'),
```

Comment:
Sorry, I'm having trouble seeing how you want this reordered. Can you order it in something like:

- classNames
- classNameBindings
- canEdit
- whateverElse

so I can understand better?